# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../dfb-extra/libmpeg3-mpeg3_open.patch
# Copyright (C) 2006 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

libmpeg3 API change since version 1.7

--- DirectFB-extra-0.9.25/interfaces/IDirectFBVideoProvider/idirectfbvideoprovider_libmpeg3.c.orig	2006-02-26 22:38:13.000000000 +0100
+++ DirectFB-extra-0.9.25/interfaces/IDirectFBVideoProvider/idirectfbvideoprovider_libmpeg3.c	2006-09-14 14:03:25.000000000 +0200
@@ -1051,6 +1051,7 @@
 {
      mpeg3_t *q;
      char    *filename;
+     int     mpeg3_error;
      
      if (!ctx->filename)
           return DFB_UNSUPPORTED;
@@ -1062,8 +1063,9 @@
           return DFB_UNSUPPORTED;
      }
 
-     q = mpeg3_open( filename );
-     if (!q) {
+     q = mpeg3_open( filename, &mpeg3_error );
+     if (!q || mpeg3_error) {
+          D_ERROR( "Libmpeg3 Provider: Error opening the video file (%d)!\n", mpeg3_error );
           D_FREE( filename );
           return DFB_UNSUPPORTED;
      }
@@ -1084,7 +1086,7 @@
 static DFBResult
 Construct( IDirectFBVideoProvider *thiz, IDirectFBDataBuffer *buffer )
 {
-     int i;
+     int i, mpeg3_error;
      IDirectFBDataBuffer_data *buffer_data;
 
      DIRECT_ALLOCATE_INTERFACE_DATA(thiz, IDirectFBVideoProvider_Libmpeg3)
@@ -1103,7 +1105,7 @@
 
 
      /* open mpeg3 file */
-     data->file          = mpeg3_open( data->filename );
+     data->file          = mpeg3_open( data->filename, &mpeg3_error );
 
      /* fetch information about video */
      data->video.width   = mpeg3_video_width( data->file, 0 );
